e2e/testcases/reconciler_manager_test.go

```diff
@@ -603,3 +603,44 @@ func TestAutopilotReconcilerAdjustment(t *testing.T) {
 	})
 
 }
+
+func TestAllowVerticalScaleResourceAdjustment(t *testing.T) {
+	nt := nomostest.New(t, ntopts.SkipMonoRepo)
+
+	reconcilerDeployment := &appsv1.Deployment{}
+	if err := nt.Validate(nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem, reconcilerDeployment); err != nil {
+		nt.T.Fatal(err)
+	}
+	rs := &v1beta1.RootSync{}
+	if err := nt.Get(configsync.RootSyncName, configsync.ControllerNamespace, rs); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	allowVerticalScale := false
+	for _, c := range reconcilerDeployment.Spec.Template.Spec.Containers {
+		if c.Name == "reconciler-manager" {
+			for _, arg := range c.Args {
+				if arg == "--allow-vertical-scale" {
+					allowVerticalScale = true
+				}
+			}
+		}
+	}
+
+	var updatedFirstContainerCPURequest int64
+	var updatedFirstContainerMemoryRequest int64
+	firstContainerName := reconcilerDeployment.Spec.Template.Spec.Containers[0].Name
+	updatedFirstContainerCPURequest = 159
+	updatedFirstContainerMemoryRequest = 259
+
+	if allowVerticalScale {
+		nt.MustMergePatch(rs, fmt.Sprintf(`{"spec":{"override":{"resources":[{"containerName":"%s","memoryRequest":"%dMi", "cpuRequest":"%dm"}]}}}`,
+			firstContainerName, updatedFirstContainerMemoryRequest, updatedFirstContainerCPURequest))
+	}
+
+	nomostest.Wait(nt.T, "Waiting for fourth reconciliation", nt.DefaultWaitTimeout, func() error {
+		return nt.Validate(nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem, &appsv1.Deployment{},
+			hasGeneration(3), firstContainerCPURequestIs(updatedFirstContainerCPURequest),
```

Comment:
Add a comment to explain why generation is 3.

Reply:
generation should be 2, added explanation comment

Reply:
[-2] Is generation 2 or 3? It is changed to 3 in your latest commit now. I didn't see explanation comment either.

Reply:
Why is this comment marked as resolved? What is the expected generation?

Reply:
as explained in https://github.com/GoogleContainerTools/kpt-config-sync/pull/20#discussion_r956371505, the test works differently in the earlier commits, therefore the expected generation was different

Reply:
unresolving for now since this is related with https://github.com/GoogleContainerTools/kpt-config-sync/pull/20#discussion_r956371505

Reply:
ack